<?xml version="1.0"?>
<sdf version="1.9">
  <model name="pine_tree">
    <static>true</static>
    <link name="link">
      <collision name="collision">
        <geometry>
          <mesh>
            <uri>model://pine_tree/meshes/pine_tree.dae</uri>
          </mesh>
        </geometry>
        <surface>
          <contact>
            <ode>
              <kp>1000000.0</kp>
              <kd>100.0</kd>
            </ode>
          </contact>
          <friction>
            <ode>
              <mu>0.8</mu>
              <mu2>0.8</mu2>
            </ode>
          </friction>
        </surface>
      </collision>

      <visual name="bark_visual">
        <geometry>
          <mesh>
            <uri>model://pine_tree/meshes/pine_tree.dae</uri>
            <submesh>
              <name>Bark</name>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <ambient>1.0 1.0 1.0 1</ambient>
          <diffuse>1.0 1.0 1.0 1</diffuse>
          <specular>0.1 0.1 0.1 1</specular>
          <pbr>
            <metal>
              <albedo_map>model://pine_tree/materials/textures/bark_diffuse.png</albedo_map>
            </metal>
          </pbr>
        </material>
      </visual>
      
      <visual name="branch_visual">
        <geometry>
          <mesh>
            <uri>model://pine_tree/meshes/pine_tree.dae</uri>
            <submesh>
              <name>Branch</name>
            </submesh>
          </mesh>
        </geometry>
        <material>
          <ambient>1.0 1.0 1.0 1</ambient>
          <diffuse>1.0 1.0 1.0 1</diffuse>
          <specular>0.1 0.1 0.1 1</specular>
          <pbr>
            <metal>
              <albedo_map>model://pine_tree/materials/textures/branch_2_diffuse.png</albedo_map>
            </metal>
          </pbr>
        </material>
      </visual>
    </link>
  </model>
</sdf>